### PR TITLE
Add custom path to the build directory

### PR DESCRIPTION
**Pull request by nk0leg**

### CODE REVIEW

**Comment by malirod** [comment]
@nk0leg, fix the commit message(amend). "Added" -> "Add"

File: ms/common_setup.cmd [hunk 1]
```diff
@@ -1,13 +1,19 @@
 @echo off
+setlocal EnableDelayedExpansion
 
+set CUSTOM_BUIL_DIR=%~1
+if "%CUSTOM_BUIL_DIR%"=="" (
```
Comment (anosach-luxoft) [inline ms/common_setup.cmd]: why not `!CUSTOM_BUIL_DIR!`? It's local variable too

File: ms/common_setup.cmd [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 set BUILD_DIR="%SDL_ROOT%\build"
-set SDL_BUILD_DIR="%BUILD_DIR%\%SDL_BUILD%"
-if not exist %SDL_BUILD_DIR% mkdir %SDL_BUILD_DIR%
-if not exist %SDL_BUILD_DIR% goto error
+) else (
+  set BUILD_DIR=%CUSTOM_BUIL_DIR%
```
Comment (anosach-luxoft) [inline ms/common_setup.cmd]: again

File: ms/common_setup.cmd (hunk 1 above)
Comment (malirod) [inline ms/common_setup.cmd]: %VAR% - used when evaluation doesn't take place.
!VAR! -used when var value is calculated(related to the setlocal EnableDelayedExpansion)

File: ms/common_setup.cmd [hunk 3]
```diff
@@ -1,13 +1,19 @@
 @echo off
+setlocal EnableDelayedExpansion
 
+set CUSTOM_BUIL_DIR=%~1
+if "%CUSTOM_BUIL_DIR%"=="" (
 set BUILD_DIR="%SDL_ROOT%\build"
```
Comment (anosach-luxoft) [inline ms/common_setup.cmd]: indent

Comment (nk0leg) [inline ms/common_setup.cmd]: see 9ebec2c

---

This e-mail and any attachment(s) are intended only for the recipient(s) named above and others who have been specifically authorized to receive them. They may contain confidential information. If you are not the intended recipient, please do not read this email or its attachment(s). Furthermore, you are hereby notified that any dissemination, distribution or copying of this e-mail and any attachment(s) is strictly prohibited. If you have received this e-mail in error, please immediately notify the sender by replying to this e-mail and then delete this e-mail and any attachment(s) or copies thereof from your system. Thank you.

**Comment by anosach-luxoft** [comment]
Reviewed

**Comment by malirod** [comment]
Seems this PR breaks default logic. During my test the setup scripts are trying to rebuild openssl even if I already have it. Check this, please.

**Comment by malirod** [comment]
@nk0leg, why you cannot use symlinks instead of passing come custom path? I mean if you prefer some other place for build dir, why not to just create the symlink?

**Comment by nk0leg** [comment]
I'll try to find another solution.
Work on this task temporarily suspended

---

This e-mail and any attachment(s) are intended only for the recipient(s) named above and others who have been specifically authorized to receive them. They may contain confidential information. If you are not the intended recipient, please do not read this email or its attachment(s). Furthermore, you are hereby notified that any dissemination, distribution or copying of this e-mail and any attachment(s) is strictly prohibited. If you have received this e-mail in error, please immediately notify the sender by replying to this e-mail and then delete this e-mail and any attachment(s) or copies thereof from your system. Thank you.

**Pull request closed without merging**